L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -0,0 +1,772 @@
+
+  eventnum = 0;
+  if (asciiEventOutName_ != "") {
+    asciiEventOut_.open(asciiEventOutName_.c_str());
+  }
+
+  if (settings.debugTracklet()) {
+    edm::LogVerbatim("Tracklet") << "cabling DTC links :     " << DTCLinkFile.fullPath();
+    edm::LogVerbatim("Tracklet") << "module cabling :     " << moduleCablingFile.fullPath();
+    edm::LogVerbatim("Tracklet") << "DTC link layer disk :     " << DTCLinkLayerDiskFile.fullPath();
+
+    edm::LogVerbatim("Tracklet") << "fit pattern :     " << fitPatternFile.fullPath();
+    edm::LogVerbatim("Tracklet") << "process modules : " << processingModulesFile.fullPath();
+    edm::LogVerbatim("Tracklet") << "memory modules :  " << memoryModulesFile.fullPath();
+    edm::LogVerbatim("Tracklet") << "wires          :  " << wiresFile.fullPath();
+  }
+}
+
+/////////////
+// DESTRUCTOR
+L1FPGATrackProducer::~L1FPGATrackProducer() {
+  if (asciiEventOutName_ != "") {
+    asciiEventOut_.close();
+  }
+}
+
+//////////
+// END JOB
+void L1FPGATrackProducer::endRun(const edm::Run& run, const edm::EventSetup& iSetup) {}
+
+void L1FPGATrackProducer::endJob() {
+#ifdef USEHYBRID
+#ifdef USE_HLS
+  TMTT::Settings settingsTMTT;
+  TMTT::KFParamsCombCallHLS fitterKF(&settingsTMTT, settings.nHelixPar(), "KFfitterHLS");
+  fitterKF.endJob();  // Check bits ranges of KF HLS.
+#endif
+#endif
+}
+
+////////////
+// BEGIN JOB
+void L1FPGATrackProducer::beginRun(const edm::Run& run, const edm::EventSetup& iSetup) {}
+
+//////////
+// PRODUCE
+void L1FPGATrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  typedef std::map<Trklet::L1TStub,
+                   edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_> >, TTStub<Ref_Phase2TrackerDigi_> >,
+                   L1TStubCompare>
+      stubMapType;
+  typedef edm::Ref<edmNew::DetSetVector<TTCluster<Ref_Phase2TrackerDigi_> >, TTCluster<Ref_Phase2TrackerDigi_> >
+      TTClusterRef;
+
+  /// Prepare output
+  std::unique_ptr<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > L1TkTracksForOutput(
+      new std::vector<TTTrack<Ref_Phase2TrackerDigi_> >);
+
+  stubMapType stubMap;
+
+  /// Geometry handles etc
+  edm::ESHandle<TrackerGeometry> geometryHandle;
+
+  /// Set pointers to Stacked Modules
+  iSetup.get<TrackerDigiGeometryRecord>().get(geometryHandle);
+
+  ////////////////////////
+  // GET MAGNETIC FIELD //
+  edm::ESHandle<MagneticField> magneticFieldHandle;
+  iSetup.get<IdealMagneticFieldRecord>().get(magneticFieldHandle);
+  const MagneticField* theMagneticField = magneticFieldHandle.product();
+  double mMagneticFieldStrength = theMagneticField->inTesla(GlobalPoint(0, 0, 0)).z();
+
+  ////////////
+  // GET BS //
+  edm::Handle<reco::BeamSpot> beamSpotHandle;
+  iEvent.getByToken(bsToken_, beamSpotHandle);
+  math::XYZPoint bsPosition = beamSpotHandle->position();
+
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  iSetup.get<TrackerDigiGeometryRecord>().get(tGeomHandle);
+
+  eventnum++;
+  Trklet::SLHCEvent ev;
```

Comment:
we have removed the "using namespace trklet" from the producer (to avoid confusion of eg Stub, Setting, and similar classes that could exist also in other namespaced).